packages/liveblocks-core/src/connection.ts

```diff
@@ -261,10 +263,55 @@ function log(level: LogLevel, message: string) {
   };
 }
 
+function logPrematureErrorOrCloseEvent(e: IWebSocketEvent | Error) {
+  // Produce a useful log message
+  const conn = "Connection to Liveblocks websocket server";
+  return (ctx: Readonly<Context>) => {
+    if (e instanceof Error) {
+      console.warn(`${conn} could not be established. ${String(e)}`);
+    } else {
+      console.warn(
+        isCloseEvent(e)
+          ? `${conn} closed prematurely (code: ${e.code}). Retrying in ${ctx.backoffDelay}ms.`
+          : `${conn} could not be established.`
+      );
+    }
+  };
+}
+
+function logCloseEvent(event: IWebSocketCloseEvent) {
+  return (ctx: Readonly<Context>) => {
+    console.warn(
+      `Connection to Liveblocks websocket server closed (code: ${event.code}). Retrying in ${ctx.backoffDelay}ms.`
+    );
+  };
+}
+
+const logPermanentClose = log(
+  LogLevel.WARN,
+  "Connection to WebSocket closed permanently. Won't retry."
+);
```

Comment:
This just DRYs up logs that already exist, but were inlined in the machine's state definitions below.